lib/ansible/modules/cloud/vmware/vmware_dvswitch.py

```diff
@@ -1,76 +1,88 @@
-        required: False
-        version_added: 2.5
+      description:
+      - The version of the distributed vSwitch to create.
+      - Can be 6.5.0, 6.0.0, 5.5.0, 5.1.0, 5.0.0 with a vCenter running vSphere 6.5.
+      - Needed if you have a vcenter version > ESXi version to join DVS. If not specified version=version of vCenter.
+      - Required only if C(state) is set to C(present).
+      version_added: 2.5
+      choices: ['5.0.0', '5.1.0', '5.5.0', '6.0.0', '6.5.0']
     mtu:
-        description:
-            - The switch maximum transmission unit
-        required: True
+      description:
+      - The switch maximum transmission unit.
+      - Required parameter for C(state) both C(present) and C(absent), before Ansible 2.6 version.
+      - Required only if C(state) is set to C(present), for Ansible 2.6 and onwards.
+      - Accepts value between 1280 to 9000 (both inclusive).
     uplink_quantity:
-        description:
-            - Quantity of uplink per ESXi host added to the switch
-        required: True
+      description:
+      - Quantity of uplink per ESXi host added to the distributed vSwitch.
+      - Required parameter for C(state) both C(present) and C(absent), before Ansible 2.6 version.
+      - Required only if C(state) is set to C(present), for Ansible 2.6 and onwards.
     discovery_proto:
-        description:
-            - Link discovery protocol between Cisco and Link Layer discovery
-        choices:
-            - 'cdp'
-            - 'lldp'
-        required: True
+      description:
+      - Link discovery protocol between Cisco and Link Layer discovery.
+      - Required parameter for C(state) both C(present) and C(absent), before Ansible 2.6 version.
+      - Required only if C(state) is set to C(present), for Ansible 2.6 and onwards.
+      choices: ['cdp', 'lldp']
     discovery_operation:
-        description:
-            - Select the discovery operation
-        choices:
-            - 'both'
-            - 'none'
-            - 'advertise'
-            - 'listen'
+      description:
+      - Select the discovery operation.
+      - Required parameter for C(state) both C(present) and C(absent), before Ansible 2.6 version.
+      - Required only if C(state) is set to C(present), for Ansible 2.6 and onwards.
+      choices: ['both', 'none', 'advertise', 'listen']
     state:
-        description:
-            - Create or remove dvSwitch
-        default: 'present'
-        choices:
-            - 'present'
-            - 'absent'
-        required: False
+      description:
+      - If set to C(present) and distributed vSwitch does not exists then distributed vSwitch is created.
+      - If set to C(absent) and distributed vSwitch exists then distributed vSwitch is deleted.
+      - Update distributed vSwitch is not supported.
+      default: 'present'
+      choices: ['present', 'absent']
+    vendor:
+      description:
+      - Name of the vendor.
```

Comment:
Do we know other supported vendors? (Don't think there are others than VMware currently in vSphere 6.7, but there were others like Cisco and IBM in the past)

Reply:
I would remove that parameter to avoid confusion and just set spec.productInfo.vendor to 'VMware, Inc.'.

Reply:
This fix come in picture due to this change only. Before change we are hardcoding `VMware` value for vendor. PR allows user to specify there version of vendor rather than hardcoding value. I am not sure about all allowed values of vendor so it is kept like this.